tests/scalers/cpu.test.ts

```diff
@@ -0,0 +1,130 @@
+  sh.config.silent = true
+  sh.exec(`kubectl create namespace ${testNamespace}`)
+
+  fs.writeFileSync(deployMentFile.name, deployMentYaml)
+  t.is(
+    0,
+    sh.exec(`kubectl apply -f ${deployMentFile.name} --namespace ${testNamespace}`).code,
+    'Deploying php deployment should work.'
+  )
+  t.is(0, sh.exec(`kubectl rollout status deploy/php-apache -n ${testNamespace}`).code, 'Deployment php rolled out succesfully')
+})
+
+test.serial('Deployment should have 1 replicas on start', t => {
+  const replicaCount = sh.exec(
+    `kubectl get deployment.apps/php-apache --namespace ${testNamespace} -o jsonpath="{.spec.replicas}"`
+  ).stdout
+  t.is(replicaCount, '1', 'replica count should start out as 1')
+})
+
+test.serial(`Creating Job should work`, async t => {
+  fs.writeFileSync(triggerFile.name, triggerJob)
+  t.is(
+    0,
+    sh.exec(`kubectl apply -f ${triggerFile.name} --namespace ${testNamespace}`).code,
+    'creating job should work.'
+  )
+})
+
+test.serial(`Deployment should scale to 5 after 3 minutes`, async t => {
+  const maxReplicaCount = '5'
+  // check replicacount on constant triggering :
+  t.true(await waitForDeploymentReplicaCount(5, 'php-apache', testNamespace, 18, 10000), 'Replica count should be 5 after 180 seconds')
+})
+test.serial(`Deployment should scale back to 1 after 10 minutes`, async t => {
+  // check for the scale down :
+  t.true(await waitForDeploymentReplicaCount(1, 'php-apache', testNamespace, 60, 10000), 'Replica count should be 1 after 10 minutes')
+})
+
+test.after.always.cb('clean up workload test related deployments', t => {
+  const resources = [
+    'deployment.apps/php-apache',
+    'jobs.batch/trigger-job',
+  ]
+  for (const resource of resources) {
+    sh.exec(`kubectl delete ${resource} --namespace ${testNamespace}`)
+  }
+  sh.exec(`kubectl delete namespace ${testNamespace}`)
+  t.end()
+})
+
+const deployMentYaml = `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: php-apache
+spec:
+  selector:
+    matchLabels:
+      run: php-apache
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        run: php-apache
+    spec:
+      containers:
+      - name: php-apache
+        image: k8s.gcr.io/hpa-example
+        ports:
+        - containerPort: 80
+        resources:
+          limits:
+            cpu: 500m
+          requests:
+            cpu: 200m
+        imagePullPolicy: IfNotPresent
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: php-apache
+  labels:
+    run: php-apache
+spec:
+  ports:
+  - port: 80
+  selector:
+    run: php-apache
+---
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: cpu-scaledobject
+  labels:
+    run: php-apache
+spec:
+  scaleTargetRef:
+    name: php-apache
+  triggers:
+  - type: cpu
+    metadata:
+      type: Utilization
+      value: "50"`
+const triggerJob = `apiVersion: batch/v1
+kind: Job
+metadata:
+  name: trigger-job
+spec:
+  template:
+    spec:
+      containers:
+      - image: jordi/ab
```

Comment:
Do we need to use this custom image? Can't we use some standar one, since we are just running bash?

Reply:
@zroubalik  Yes, we can change it. ill add another one I've tested on.

Reply:
for example this one: https://hub.docker.com/_/bash

but feel free to add whatever you like

Reply:
Oh yes we can use this one too, for now i have already tested on another one. PTAL